### PR TITLE
core: add arrayList tests

### PR DESCRIPTION
**Pull request by louisgreiner**

### CODE REVIEW

File: core/kt-fast-collections/src/test/kotlin/TestArrayList.kt [hunk 1]
```diff
@@ -32,10 +334,51 @@ class ArrayListTest {
...
+        val arrayList1 = MutableIntArrayList()
+        assertEquals(true, arrayList1.add(2))
+        assertEquals(true, arrayList1.add(3))
+        assertEquals(true, arrayList1.add(1))
+        val arrayList2 = MutableIntArrayList()
+        assertEquals(true, arrayList2.add(1))
+        assertEquals(true, arrayList2.add(2))
+        assertEquals(true, arrayList2.add(3))
+
+        assertEquals(!true, arrayList1 == arrayList2)
```
Comment (shenriotpro) [inline core/kt-fast-collections/src/test/kotlin/TestArrayList.kt]: I'm guessing there is a name for !true :)

File: core/kt-fast-collections/src/test/kotlin/TestArrayList.kt [hunk 2]
```diff
@@ -32,10 +334,51 @@ class ArrayListTest {
...
+        val arrayList2 = MutableIntArrayList()
+        assertEquals(true, arrayList2.add(1))
+        assertEquals(true, arrayList2.add(2))
+        assertEquals(true, arrayList2.add(3))
+
+        assertEquals(true, arrayList1 == arrayList2)
+    }
+
+    @Test
+    fun `lists with same elements and different order should be equal`() {
```
Comment (shenriotpro) [inline core/kt-fast-collections/src/test/kotlin/TestArrayList.kt]: Is that a typo?

**Review by shenriotpro — CHANGES_REQUESTED** [review]
Great, thanks

File: core/kt-fast-collections/src/test/kotlin/TestArrayList.kt (hunk 2 above)
Comment (louisgreiner) [reply core/kt-fast-collections/src/test/kotlin/TestArrayList.kt]: true, fixed ^^

File: core/kt-fast-collections/src/test/kotlin/TestArrayList.kt [hunk 3]
```diff
@@ -32,10 +334,51 @@ class ArrayListTest {
...
+        val arrayList1 = MutableIntArrayList()
+        assertEquals(true, arrayList1.add(1))
+        assertEquals(true, arrayList1.add(2))
+        assertEquals(true, arrayList1.add(3))
+        val arrayList2 = MutableIntArrayList()
+        assertEquals(true, arrayList2.add(1))
+        assertEquals(true, arrayList2.add(2))
+        assertEquals(true, arrayList2.add(3))
+
+        assertEquals(true, arrayList1 == arrayList2)
```
Comment (shenriotpro) [inline core/kt-fast-collections/src/test/kotlin/TestArrayList.kt]: perhaps we could use assertEquals(arrayList1, arrayList2)

File: core/kt-fast-collections/src/test/kotlin/TestArrayList.kt [hunk 4]
```diff
@@ -32,10 +334,51 @@ class ArrayListTest {
...
+        val arrayList1 = MutableIntArrayList()
+        assertEquals(true, arrayList1.add(2))
+        assertEquals(true, arrayList1.add(3))
+        assertEquals(true, arrayList1.add(1))
+        val arrayList2 = MutableIntArrayList()
+        assertEquals(true, arrayList2.add(1))
+        assertEquals(true, arrayList2.add(2))
+        assertEquals(true, arrayList2.add(3))
+
+        assertEquals(false, arrayList1 == arrayList2)
```
Comment (shenriotpro) [inline core/kt-fast-collections/src/test/kotlin/TestArrayList.kt]: assertNotEquals may be useful https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/

**Review by shenriotpro — APPROVED** [review]

**Review by Khoyo — APPROVED** [review]
We should change assert to assertTrue()

Otherwise LGTM

**Pull request merged**